backstory/src/demo/java/com/google/sps/servlets/PerspectiveServlet.java

```diff
@@ -39,11 +46,90 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     response.setContentType("application/json;");
     Gson gson = new Gson();
 
-    String[] cannedOutput = new String[0];
+    String json = request.getReader().readLine();
 
-    // write canned output to servlet
-    String json = gson.toJson(cannedOutput);
+    
+    // get the text to be analyzedfrom the JSON & handle error if it cannot be converted
+    String text = null;
+    try {
+      JSONObject jsonObject = new JSONObject(json);
+      text = jsonObject.getString("text");
+    } catch (JSONException exception) {
+      String errorMessage = "Could not convert text sent to server from JSON.";
 
-    response.getWriter().println(json);
+      handleError(response, HttpServletResponse.SC_BAD_REQUEST, errorMessage);
+      return;
+    }
+
+    // check that text is valid (not null or empty)
+    if (text == null || text.equals("")) {
+      handleError(response, HttpServletResponse.SC_BAD_REQUEST, "Text input was null or empty");
+      return;
+    }
+
+    // as we want to display the values as well as the decision in our demo
+    // we need to access internal code that the MVP would not
+    PerspectiveValues values = null;    
+    try {
+      PerspectiveAPIFactory factory = new PerspectiveAPIFactoryImpl();
+      PerspectiveAPI api = factory.newInstance();
+      PerspectiveAPIClient apiClient = new PerspectiveAPIClient(api);
+      values = apiClient.analyze(Arrays.asList(PerspectiveStoryAnalysisManager.REQUESTED_ATTRIBUTES), text);
```

Comment:
As mentioned in a separate comment, this will make a duplicate request (and double usage of this service). How about following up by adding a PerspectiveValues property to StoryDecision, so that this info can also be retrieved from PerspectiveManager?

Reply:
I added a PerspectiveDecision class for this purpose, which is `StoryDecision` with added `PerspectiveValues` field. As such, this info can be retrieved from PerspectiveManager, and there's only one request to the service.

Reply:
Done